fix: Improve getFileExtension() readability and handle leading dot extensions.

damanV5 opened the pull request:
## Description
More robust edge case handling for getFileExtension().

**Case** : Type is not provided explicitly in {src,type} object.
In this case getFileExtension() is called to fetch the extension so as to get the type.

For certain cases, fn does not returns the correct extension.
For Eg: 
1. https://demo.unified-streaming.com/k8s/features/stable/video/tears-of-steel/tears-of-steel.ism/.m3u8

In this case fn returns "". 
I have simplified getFileExtension() so that fn will return m3u8 in this case as well.

## Specific Changes proposed
Simplified getFileExtension() for better readability and to handle more edge cases.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chrome, Firefox, IE)
  - [x] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
  - [x] Has no DOM changes which impact accessiblilty or trigger warnings (e.g. Chrome issues tab)
  - [x] Has no changes to JSDoc which cause `npm run docs:api` to error
- [x] Reviewed by Two Core Contributors

damanV5 commented:
@gkatsev Could you please help with getting this merged?

mister-ben reviewed (dismissed):
Thanks for the PR! It's a nice simplification. It would consider `com` in `https://example.com` a file extensions too, but so does the original.

Could you add a test to test/until/utils/url.test.js with the type of URL that should now work?

I don't think "accept a training slash" is the best description. When I read that I thought it's intended for a URL like `.../file.m3u8/`. Something like to "match an extension without a filename"?

damanV5 commented:
@mister-ben  I have added test cases in `test/until/utils/url.test.js` and updated the PR description as well.
Let me know if something else needs to be done.

damanV5 commented on test/unit/utils/url.test.js:
```diff
@@ -55,6 +55,8 @@ QUnit.test('should get the file extension of the passed path', function(assert)
   assert.equal(Url.getFileExtension('http://www.test.com/video.mp4'), 'mp4');
   assert.equal(Url.getFileExtension('http://foo/bar/test.video.wgg'), 'wgg');
+  assert.equal(Url.getFileExtension('http://www.test.com/video/.mp4'), 'mp4');
+  assert.equal(Url.getFileExtension('http://www.test.com/video/.mp4/'), 'mp4');
```
@mister-ben The updated regex will also handle a trailing slash , which was not handled before.

Though, I can remove it.

damanV5 replied:
I just checked the original regex also handled the trailing slash, but there was no test case written for that.

damanV5 replied:
@mister-ben

alex-barstow replied:
@damanV5 Thx for the contribution! Would you mind adding an additional test case that has a complete file name + extension + trailing slash? ex. `'http://www.test.com/video.mp4/'`

In addition to the case of having only an extension (`'/.mp4'`), are there other specific known edge cases this change will handle? If not, then I'd recommend changing the PR title to something more specific than "handle edge cases", for example: `fix: Improve getFileExtension() readability and handle leading dot extensions`

damanV5 replied:
Sure @alex-barstow , Let me add that.

damanV5 replied:
I have added `http://www.test.com/video.mp4/` in url.test.js.

alex-barstow reviewed (changes requested):
A couple very small recommendations but otherwise LGTM! Thx @damanV5

damanV5 commented:
@alex-barstow  I have checked, this enhancement will handle the following test cases :-

 `http://www.test.com/video/.mp4`
`http://www.test.com/video/.mp4/`


-----

`http://www.test.com/video.mp4/` : This was already handled but test case was not added , so I have now added a test case.

damanV5 commented:
@alex-barstow  Sure. I have now updated the PR title and updated the base branch.

alex-barstow reviewed (approved):
Approved. Just need one more contributor to approve

damanV5 commented:
@gkatsev  / @heff  Can you please approve this?

adrums86 reviewed (approved):
Thanks for your contribution! LGTM.
